gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-account/src/main/java/io/gravitee/am/gateway/handler/account/resources/AccountFactorsEndpointHandler.java

```diff
@@ -702,6 +702,11 @@ private EnrolledFactor buildEnrolledFactor(Factor factor,
             enrolledFactor.setSecurity(new EnrolledFactorSecurity(SHARED_SECRET, enrollment.getKey(), additionalData));
         }
 
+        if (account.getExtensionPhoneNumber() != null && enrolledFactor.getChannel() != null) {
+            var additionalData = enrolledFactor.getChannel().getAdditionalData();
```

Comment:
Can we have a NPE here ?

Reply:
it shouldn't as the Channel object initialize this attribute with new HashMap. But you are right I made an update to have defensive code